src/bin/common/parse_command.ml

```diff
@@ -693,11 +693,51 @@ let parse_execution_opt =
          info ["add-parser"] ~docs ~doc) in
 
   let preludes =
+    let parse_prelude p =
+      if Sys.file_exists p then
+        if Sys.is_directory p then
+          Fmt.error "'%s' is a directory" p
+        else
+          Ok p
+      else
+        let p' = Filename.concat Config.preludesdir p in
+        if Sys.file_exists p' then begin
+          begin if String.starts_with ~prefix:"b-set-theory" p then
+              Printer.print_wrn ~header:true
+                "Support for the B set theory is deprecated since version 2.5.0@ \
+                 and may be removed in a future version. If you are actively using@ \
+                 it, please make yourself known to the Alt-Ergo developers by@ \
+                 writing to <alt-ergo@ocamlpro.com>."
+            else if String.starts_with ~prefix:"fpa-theory" p then
+              Printer.print_wrn ~header:true
+                "@[Support for the FPA theory has been integrated as a builtin @ \
+                 theory prelude in version 2.5.0 and is enabled by default. This @ \
+                 option is no longer needed, and the '%s'@ prelude will be removed@ \
+                 in a later version.@]" p
+          end;
+
+          if Sys.is_directory p' then
+            Fmt.error "'%s' is a directory" p
```

Comment:
We have a printer for error in Printers

Reply:
I don't think we do; this is an error occuring during argument parsing/setup, it should use cmdliner error mechanism rather than Alt-Ergo's

`Fmt.error` returns a `('a, string) result` which is used by cmdliner